### PR TITLE
feat: add inline rules

### PR DESCRIPTION
**Pull request by jeongshin**
Hello. 

Thanks for great library again! 

I'm currently working on customizing all markdown elements (or tags). 
However, I have one problem rendering `link`. 

## as-is

Screenshot 2023-08-09 at 9 42 06 PM

By default, links are rendered in paragraph as inline component nested in `Text` node. 
So, if I put `View` inside `Text` component, components are overlapped each other. 

(btw, I'm trying to add open graphic info under link tag) 

## to-be

I opened optional config to determine `link` tag to be `block` or `inline`. 

 
Screenshot 2023-08-09 at 9 45 10 PM

Can it be done like this?

Or can you recommend any other solution to solve this problem?

### CODE REVIEW

**Comment by gmsgowtham** [comment]
@jeongshin It appears that there is a problem with the parser. We are already handling any image nodes inside a paragraph as a block. Therefore, could you please post a reproduction of the issue so that I can access the solution that has been provided?

File: examples/react-native-marked-sample/package.json [hunk 1]
```diff
@@ -10,25 +10,25 @@
...
-    "react-native-svg": "12.3.0",
-    "react-native-web": "~0.18.7"
+    "expo": "^49.0.0",
+    "expo-splash-screen": "~0.20.5",
+    "expo-status-bar": "~1.6.0",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "react-native": "0.72.3",
+    "react-native-svg": "13.9.0",
+    "react-native-web": "~0.19.6"
```
Comment (jeongshin) [inline examples/react-native-marked-sample/package.json:20]: I upgraded expo version. expo 46 is not supported for current play store expo app.

File: examples/react-native-marked-sample/src/App.tsx [hunk 2]
```diff
@@ -62,6 +63,15 @@ class CustomRenderer extends Renderer implements RendererInterface {
...
 		return null;
 	}
+
+	link(uri: string) {
+		return (
+			<View style={{ width: "100%" }} key={this.getKey()}>
+				<Text style={{ textDecorationLine: "underline" }}>{uri}</Text>
+				<View style={{ width: "100%", height: 150, backgroundColor: "blue" }} />
+			</View>
+		);
```
Comment (jeongshin) [inline examples/react-native-marked-sample/src/App.tsx:73]: Here's the problem. 
View is nested inside the `Text` and I wish it to be in `View`. 
It's breaking UI only on android. It's looks fine on ios.

**Review by jeongshin — COMMENTED** [review]
I added examples. 
Can you take a look? 

Thank you.

File: examples/react-native-marked-sample/src/App.tsx (hunk 2 above)
Comment (gmsgowtham) [reply examples/react-native-marked-sample/src/App.tsx:73]: @jeongshin Yeah, I can understand the issue. But can you please provide a full blown repro with CodeSandbox? https://codesandbox.io/s/react-native-marked-l2hpi3?file=/src/App.js

Comment (jeongshin) [reply examples/react-native-marked-sample/src/App.tsx:73]: absolutely!

Comment (jeongshin) [reply examples/react-native-marked-sample/src/App.tsx:73]: I'm having problem with resolving error on dependencies. 

Screenshot 2023-08-10 at 2 05 24 PM

Can you check the issue through expo example app?

Comment (gmsgowtham) [reply examples/react-native-marked-sample/src/App.tsx:73]: I've updated the CodeSandbox. It should work now.

Comment (gmsgowtham) [reply examples/react-native-marked-sample/src/App.tsx:73]: https://codesandbox.io/s/react-native-marked-l2hpi3 Some dep issue.

Comment (jeongshin) [reply examples/react-native-marked-sample/src/App.tsx:73]: Thanks! 

https://codesandbox.io/s/react-native-marked-forked-j92kv4 

Here's example. 
Blue colored area is where I want to customize. 
It looks fine on web but not on android real device. 🥲

Comment (gmsgowtham) [reply examples/react-native-marked-sample/src/App.tsx:73]: @jeongshin What's the markdown that leads to this output? 

```jsx
<View style={{ width: "100%" }} key={this.getKey()}>
      <Text style={{ textDecorationLine: "underline" }}>{uri}</Text>
      <View style={{ width: "100%", height: 150, backgroundColor: "blue" }} />
</View>
```

Comment (jeongshin) [reply examples/react-native-marked-sample/src/App.tsx:73]: I'm sorry but I don't get what you mean by 'leads to this output'. 

```ts
import { ScrollView, View, Text } from "react-native";
import { Renderer, RendererInterface, useMarkdown } from "react-native-marked";

export default function App() {
  const elements = useMarkdown("[Google](https://www.google.com)", {
    renderer,
  });

  console.log(elements);

  return (
    <View style={{ width: "100%" }}>
      {elements.map((el) => {
        // console.log(el);
        return el;
      })}
    </View>
  );
}

class CustomRenderer extends Renderer implements RendererInterface {
  constructor() {
    super();
  }

  link(uri: string) {
    return (
      <ScrollView style={{ width: "100%" }} key={this.getKey()}>
        <Text style={{ textDecorationLine: "underline" }}>{uri}</Text>
        <View style={{ width: "100%", height: 150, backgroundColor: "blue" }} />
      </ScrollView>
    );
  }
}

const renderer = new CustomRenderer();
```
Screenshot 2023-08-11 at 12 39 18 AM

here's minimal code and link is parsed as 'paragraph'. 

So I'm wondering if there's a way to change

```ts
<Text selectable={true} style={[Object]}><ScrollView … /></Text>
```

to 

```ts
<View><ScrollView … /></View>
```

I don't expect this branch to be merged, but I still need to implement this anyway. 
I can use forked version if it's okay to you.

Comment (jeongshin) [reply examples/react-native-marked-sample/src/App.tsx:73]: Thanks for reply, by the way 🙌

**Pull request closed without merging**